lib/pbench/agent/tool_data_sink.py

```diff
@@ -134,21 +136,33 @@ def terminate(self):
         if self.abort_launch:
             return 0
 
-        self.run.terminate()
-        self.run.wait()
+        try:
+            self.run.terminate()
+            self.run.wait()
+        except Exception as exc:
+            self.logger.error(
+                "Failed to terminate expected collector process: '%s'", exc
+            )
+            return 0
         return 1
 
 
 class PromCollector(BaseCollector):
+    """Persistent tool data collector for tools compatible with Prometheus"""
+
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.volume = self.tool_group_dir / "prometheus"
 
     def launch(self):
 
-        if self.host_tools_dict:
+        if self.host_tools_dict and find_executable("podman"):
```

Comment:
Not sure you need this detection here, just the check for `podman`.  Down below, in the for loop, just count how many you added to the prometheus file, and then exit on line 197.